src/Service/Api.php

```diff
@@ -1798,13 +1799,35 @@ public function getAutoscalingSettingsLink(Environment $environment, bool $manag
     public function getAutoscalingSettings(Environment $environment)
     {
         try {
-            $settings = $environment->getAutoscalingSettings();
+            $result = $environment->runOperation('autoscaling', 'get');
+        } catch (EnvironmentStateException $e) {
+            if ($e->getEnvironment()->status === 'inactive') {
+                throw new EnvironmentStateException('The environment is inactive', $e->getEnvironment());
+            }
+            throw $e;
+        }
+        return new AutoscalingSettings($result->getData(), null, $environment->client);
+    }
+
+    /**
+     * Configures the autoscaling settings for the selected environment.
+     *
+     * @param Environment $environment
+     * @param AutoscalingSettings $settings
+     */
+    public function setAutoscalingSettings(Environment $environment, array $settings)
+    {
+        if (!$this->getAutoscalingSettingsLink($environment, $manage=true)) {
+            throw new EnvironmentStateException('Not enough permissions to configure autoscaling on the environment', $environment);
```

Comment:
This might not always be permissions related, this may need to be more generic-

```suggestion
            throw new EnvironmentStateException('Managing autoscaling settings is not available on the environment', $environment);
```